From fc55f8e187792746815cacf71c20f88e37da394c Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Tue, 14 Oct 2025 18:39:14 -0700
Subject: [PATCH] livesys-i3: restore the hack to fix the installer

This was left out in the conversion from kickstarts. Without this
the installer does not launch on the live image, see
https://bugzilla.redhat.com/show_bug.cgi?id=2240823 .

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 libexec/livesys/sessions.d/livesys-i3 | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/libexec/livesys/sessions.d/livesys-i3 b/libexec/livesys/sessions.d/livesys-i3
index 39d63b1..72c563a 100755
--- a/libexec/livesys/sessions.d/livesys-i3
+++ b/libexec/livesys/sessions.d/livesys-i3
@@ -16,6 +16,15 @@ sed -i 's/^#autologin-user-timeout=.*/autologin-user-timeout=0/' /etc/lightdm/li
 # set i3 as default session, otherwise login will fail
 sed -i 's/^#user-session=.*/user-session=i3/' /etc/lightdm/lightdm.conf
 
+# Fix installer launch
+# https://bugzilla.redhat.com/show_bug.cgi?id=2240823
+cat > /etc/xdg/autostart/xfce-polkit.desktop <<EOF
+[Desktop Entry]
+Type=Application
+Name=xfce-polkit
+Exec=/usr/libexec/xfce-polkit
+EOF
+
 # Show harddisk install on the desktop
 sed -i -e 's/NoDisplay=true/NoDisplay=false/' /usr/share/applications/liveinst.desktop
 mkdir /home/liveuser/Desktop
-- 
2.51.0

